lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioConfigurationFactory.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.AlluxioProperties;
+import alluxio.conf.InstancedConfiguration;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static alluxio.conf.PropertyKey.USER_CLIENT_CACHE_DIRS;
+import static alluxio.conf.PropertyKey.USER_CLIENT_CACHE_ENABLED;
+import static alluxio.conf.PropertyKey.USER_CLIENT_CACHE_PAGE_SIZE;
+import static alluxio.conf.PropertyKey.USER_CLIENT_CACHE_SIZE;
+import static alluxio.conf.PropertyKey.USER_CLIENT_CACHE_TTL_ENABLED;
+import static alluxio.conf.PropertyKey.USER_CLIENT_CACHE_TTL_THRESHOLD_SECONDS;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.filesystem.alluxio.AlluxioFileSystemCacheConfig.CACHE_DIRECTORIES;
+import static io.trino.filesystem.alluxio.AlluxioFileSystemCacheConfig.CACHE_MAX_PERCENTAGES;
+import static io.trino.filesystem.alluxio.AlluxioFileSystemCacheConfig.CACHE_MAX_SIZES;
+import static java.lang.String.format;
+import static java.lang.String.join;
+
+public class AlluxioConfigurationFactory
+{
+    private AlluxioConfigurationFactory() {}
+
+    public static AlluxioConfiguration create(AlluxioFileSystemCacheConfig config)
+    {
+        checkArgument(config.getMaxCacheSizes().isEmpty() ^ config.getMaxCacheDiskUsagePercentages().isEmpty(),
+                "Either %s or %s must be specified", CACHE_MAX_SIZES, CACHE_MAX_PERCENTAGES);
+        int size = config.getMaxCacheSizes().isEmpty() ? config.getMaxCacheDiskUsagePercentages().size() : config.getMaxCacheSizes().size();
+        checkArgument(config.getCacheDirectories().size() == size,
+                "%s and %s must have the same size", CACHE_DIRECTORIES, config.getMaxCacheSizes().isEmpty() ? CACHE_MAX_PERCENTAGES : CACHE_MAX_SIZES);
+        config.getCacheDirectories().forEach(directory -> canWrite(Path.of(directory)));
+        List<DataSize> maxCacheSizes = config.getMaxCacheSizes().isEmpty() ?
```

Comment:
This should have been added in the original Alluxio PR, but we should log a warning or throw if any of the maxCacheSizes are 0 (probably logging a warning is the most user friendly, since it avoids a cluster entering a crash loop if it restarts, environment has changed, and cache size is calculated to be zero).

Reply:
I've just moved this logic out of the module. Extracted to a separate commit